File: problems/readability/readability.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

    string text = get_string("Text: ");

    int i = 0;
    int letters = 0;
    int words = 0;
    int sentences = 0;
    while (text[i] != '\0')
    {

        char char_at_i = text[i];

        if ((char_at_i >= 97 && char_at_i <= 122) || (char_at_i >= 65 && char_at_i <= 90))
        {
            letters++;
        }
        else if (char_at_i == 32)
        {
            words++;
        }
        else if (char_at_i == 46 || char_at_i == 63 || char_at_i == 33)
        {
            sentences++;
        }

        i++;
    }
    words++;


    float level = (0.0588 * (letters * 1.0 / words * 100) - 0.296 * (sentences * 1.0 / words * 100) - 15.8);

    if (level < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (level >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", level);
    }

}

File: problems/sentimental/readability/readability.py
# TODO
from cs50 import get_string


text = get_string("Text: ")

text = text.lower()

letters = 0
words = 0
sentences = 0

for i in range(len(text)):
    c = text[i]
    if c == " ":
        words += 1
    elif c == "." or c == "!" or c == "?":
        sentences += 1
    elif c.isalpha():
        letters += 1

words += 1

averiage_letters = (letters / words) * 100
averiage_sentences = (sentences / words) * 100


grade = 0.0588 * averiage_letters - 0.296 * averiage_sentences - 15.8

# print(averiage_letters)
# print(averiage_sentences)
# print(letters)
# print(words)
# print(sentences)
# print(grade)
# print(text)


grade = round(grade)


if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade: {grade}")

